students/wachulec_maria/lesson_10_organizing_files/filling_in_the_gaps.py

```diff
@@ -0,0 +1,67 @@
+import os
+import re
+import shutil
+
+
+def search_prefix_in_files_names(prefix):
+    all_groups_list, list_of_numbers, names_list, result = [], [], [], None
+    for file_name in os.listdir(PATH):
+        result = re.search(re.compile('((' + prefix + ')(\d\d\d)(.*?)$)'),
+                           file_name)
+        if result is None:
+            continue
+        names_list.append(result.group(1))
+        all_groups_list.append(list(result.groups()))
+        list_of_numbers.append(result.group(3))
+    if len(list_of_numbers) == 0:
+        raise AttributeError("There are no file in the path")
+    return len(result.group(3)), int(min(list_of_numbers)), sorted(
+        names_list, reverse=False), all_groups_list
+
```

Comment:
kind of overcomplicated return.
When you just look at this function, it is not clear what the return values exactly are.